### PR TITLE
Feature/form builder locked content rich text

### PR DESCRIPTION
**Pull request by timarney**
# Summary | Résumé

Adds "locked" RichText areas for introduction content and privacy policy content.

This PR also updates the current Rich Text area --- removing the modal and more button from the Panel Actions.
 
Screen Shot 2022-09-21 at 12 44 43 PM

# Test instructions | Instructions pour tester la modification

- Visit the form builder
- Note the builder now shows the introduction rich text field
- Add an element
- Note the privacy policy and confirmation rich text editors are available.  

# Unresolved questions / Out of scope | Questions non résolues ou hors sujet

- This PR adds introduction and privacyPolicy nodes to the current JSON schema... it doesn't update the JSON validation or the form preview.
- There's an existing but where the Rich text editor doesn't reload content into the Rich text editor when moving to pages such as "JSON" or form preview this PR doesn't fix that.

# Pull Request Checklist

Please complete the following items in the checklist before you request a review:

- [x] Have you completely tested the functionality of change introduced in this PR? Is the PR solving the problem it's meant to solve within the scope of the related issue?
- [x] The PR does not introduce any new issues such as failed tests, console warnings or new bugs.
- [ ] If this PR adds a package have you ensured its licensed correctly and does not add additional security issues?
- [ ] Is the code clean, readable and maintainable? Is it easy to understand and comprehend.
- [ ] Does your code have adequate comprehensible comments? Do new functions have [docstrings](https://tsdoc.org/)?
- [ ] Have you modified the change log and updated any relevant documentation?
- [ ] Is there adequate test coverage? Both unit tests and end-to-end tests where applicable?
- [ ] If your PR is touching any UI is it accessible? Have you tested it with a screen reader? Have you tested it with automated testing tools such as axe?

### CODE REVIEW

File: components/form-builder/panel/ElementPanel.tsx [hunk 1]
```diff
@@ -438,13 +442,13 @@ export const ElementWrapper = ({ item }: { item: ElementTypeWithIndex }) => {
           </ModalButton>
         )}
       >
-        {modals[item.index] && (
+        {!isRichText && modals[item.index] ? (
```
Comment (jeberhardt) [inline components/form-builder/panel/ElementPanel.tsx:445]: The previous usage of `&&` is preferred, as it omits the unecessary ` : null` as the other part of the ternary.

Comment (timarney) [reply components/form-builder/panel/ElementPanel.tsx]: Update in https://github.com/cds-snc/platform-forms-client/pull/1056/commits/fb321043948986e1562815a208bc4f77dc2ad1c5

File: components/form-builder/panel/ElementPanel.tsx [hunk 2]
```diff
@@ -466,29 +470,58 @@ const ElementPanelDiv = styled.div`
...
+    {
+      type: "paragraph",
+      children: [{ text: "Thank you for participating in the ice cream survey!" }],
+    },
+  ];
+
+  const policyTextPlaceholder = [
+    {
+      type: "paragraph",
+      children: [{ text: "All email addresses will be not be shared." }],
```
Comment (pcraig3) [inline components/form-builder/panel/ElementPanel.tsx:492]: Too many "be"s 🐝

File: components/form-builder/panel/ElementPanel.tsx [hunk 3]
```diff
@@ -466,29 +470,58 @@ const ElementPanelDiv = styled.div`
...
+    },
+  ];
 
   return (
     <ElementPanelDiv>
+      <RichTextLocked
+        addElement={true}
+        initialValue={introTextPlaceholder}
+        schemaProperty="introduction"
```
Comment (pcraig3) [inline components/form-builder/panel/ElementPanel.tsx:501]: Do we add these later to the schema, or do we not need to just yet?

File: components/form-builder/panel/ElementPanel.tsx [hunk 4]
```diff
@@ -438,7 +442,7 @@ export const ElementWrapper = ({ item }: { item: ElementTypeWithIndex }) => {
           </ModalButton>
         )}
       >
-        {modals[item.index] && (
+        {!isRichText && modals[item.index] && (
           <ModalForm
             item={item}
             properties={modals[item.index]}
```
Comment (pcraig3) [inline components/form-builder/panel/ElementPanel.tsx:448]: I'm not sure if it's here or elsewhere, but changing a question field to `richText` causes a console error.

Screen Shot 2022-09-21 at 13 21 34

File: components/form-builder/panel/PanelActionsLocked.tsx [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+      </Label>
+      {addElement && (
+        <AddButtonWrapper>
+          <FancyButton
+            onClick={() => {
+              add(0);
+            }}
+          >
+            {t("Add element")}
+          </FancyButton>
```
Comment (pcraig3) [inline components/form-builder/panel/PanelActionsLocked.tsx:62]: Not sure what's going on here, but the "Add element" button on the locked intro will always add a field to the number 2 position once there is at least one element.

You can always use Move up/down buttons to move them around, but it seems odd that it doesn't insert it into first place if you press the very first "Add element" button.

**Review by pcraig3 — COMMENTED** [review]
Left a few comments, just noticed very minor things. Have a read through them!  

Overall, this is nearly there I think.

File: components/form-builder/panel/PanelActionsLocked.tsx (hunk 5 above)
Comment (timarney) [reply components/form-builder/panel/PanelActionsLocked.tsx:62]: Maybe I'll add a prop for the index --- 

1) Only the intro has the add element button for the others I'm passing the button doesn't show ...

```
<RichTextLocked
            addElement={false}
```

Perhaps in the future we'll have another locked panel that has addElement so I'll update

File: components/form-builder/panel/ElementPanel.tsx (hunk 3 above)
Comment (timarney) [reply components/form-builder/panel/ElementPanel.tsx:501]: Will update the schema on we get the saving / reloading working in a follow-up PR once we sort out the editor.

File: components/form-builder/panel/ElementPanel.tsx (hunk 2 above)
Comment (timarney) [reply components/form-builder/panel/ElementPanel.tsx]: Updated https://github.com/cds-snc/platform-forms-client/pull/1056/commits/68a8ad92b0e3e401a9724da30125141498f1defc

File: components/form-builder/panel/PanelActionsLocked.tsx (hunk 5 above)
Comment (timarney) [reply components/form-builder/panel/PanelActionsLocked.tsx]: Didn't fully see what you were saying but was chatting with Dave and came up with a fix for now.  The add function adds 1 to the index so it was always hitting `2`.


https://github.com/cds-snc/platform-forms-client/pull/1056/commits/fa43e86736bbc562d61086c90397b1bd34361167

We don't have a use case for other indexes at the moment so I'll leave further updates for later if that works for you.

File: components/form-builder/panel/ElementPanel.tsx (hunk 4 above)
Comment (timarney) [reply components/form-builder/panel/ElementPanel.tsx:448]: 🤔 --- I don't see a way to add an optional prop type --- some online says to use `any`  

https://github.com/cds-snc/platform-forms-client/pull/1056/commits/992047a4bc6ea2fb8df2c7788aa6d8ce19d6421e

Comment (timarney) [reply components/form-builder/panel/ElementPanel.tsx:448]: In this case we don't need the modal so no children

**Review by pcraig3 — APPROVED** [review]
👍  looks good, once this is in, I can rebase my open PR

**Pull request merged**